src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1461 @@
+    int select;
+    uint8_t numInputPorts = 0;
+    
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+    printf("\n**********Please select port***********\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+
+    if(select <= 0 || select >numInputPorts)
+    {
+      printf("\n invalid port selected\n");
+      goto exit;
+    }
+   
+    port = select;
+   
+    ret = dsGetHDMISPDInfo(port, spdinfo); 
+    if(ret != dsERR_NONE)
+    {
+       UT_LOG_ERROR("Result: Failed dsGetHDMISPDInfo Error:[%s]",ut_control_GetMapString(dsError_mapTable, ret));
+    } else {
+          /* UT_LOG_INFO("Result : Passed dsGetHDMISPDInfo porttype: %s, portnumber: %s,\t"
+                           "pkttype: %d, version: %d, length: %d, rsd: %d, checksum: %d\t"
+                           "vendor_name: %s, product_des: %s, source_info: %d",
+                           ut_control_GetMapString(dsHdmiInPort_mapTable, port), port,
+                           spdinfo.pkttype, spdinfo.version, spdinfo.length, spdinfo.rsd,
+                           spdinfo.checksum, spdinfo.vendor_name, spdinfo.product_des,
+                           spdinfo.source_info);*/
+    }
+
+exit:
+UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to set edidversion.
+*
+* This test function is to set edidversion of HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 011@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_set_edidversion(void)
+{
+    gTestID = 11;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    int select;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_NONE;
+    tv_hdmi_edid_version_t edidver;
+    uint8_t numInputPorts = 0;
+
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+    printf("\n**********Please select port***********\n");
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+
+    if(select <= 0 || select >numInputPorts)
+    {
+      printf("\n invalid port selected\n");
+      goto exit;
+    }
+
+    port = select;
+  
+    printf("\n*********Please select from availabe edid versions are***********\n");
+    for(tv_hdmi_edid_version_t i = HDMI_EDID_VER_14 ; i < HDMI_EDID_VER_MAX ; i++)
+    {
+           UT_LOG_INFO("%s : %d", ut_control_GetMapString(tv_hdmi_edid_version_mapTable, i) , i);
+    }
+    printf("\n**************************************************************\n");
+
+    scanf("%d", &select);
+    readAndDiscardRestOfLine(stdin);
+
+    if(select < HDMI_EDID_VER_14 || select > HDMI_EDID_VER_MAX)
+    {
+      printf("\n invalid edid selected\n");
+      goto exit;
+    }
+    
+    edidver = (tv_hdmi_edid_version_t)select;
+        
+    ret = dsSetEdidVersion(port , edidver);
+    if(ret != dsERR_NONE)
```

Comment:
Remove the else case

In all cases, you need to see the results output

You've got some serious debugging issues todo if this was to assert and a failure of L1-L2 to fix in that case.

Reply:
done